### PR TITLE
[groheondus] Upgrade java API to latest version

### PR DESCRIPTION
**Pull request by FlorianSW**
This is a pre-step for supporting more GROHE devices, see #4550 for
that. It also enables us to use a more reliable way to login with
username/password (even given it relies on the web-based form :P).

Signed-off-by: Florian <florian.schmidt.welzow@t-online.de>

### CODE REVIEW

File: bundles/org.openhab.binding.groheondus/src/main/java/org/openhab/binding/groheondus/internal/discovery/GroheOndusDiscoveryService.java [hunk 1]
```diff
@@ -82,10 +79,10 @@ protected void startScan() {
             ThingUID bridgeUID = bridgeHandler.getThing().getUID();
             ThingUID thingUID = null;
             switch (appliance.getType()) {
-                case SenseGuardAppliance.TYPE:
+                case org.grohe.ondus.api.model.guard.Appliance.TYPE:
                     thingUID = new ThingUID(THING_TYPE_SENSEGUARD, appliance.getApplianceId());
                     break;
-                case SenseAppliance.TYPE:
+                case Appliance.TYPE:
```
Comment (cpmeister) [inline bundles/org.openhab.binding.groheondus/src/main/java/org/openhab/binding/groheondus/internal/discovery/GroheOndusDiscoveryService.java]: Given that they are two `Appliance` classes in the api I would suggest making the namespaces explicit.
```suggestion
                case org.grohe.ondus.api.model.sense.Appliance.TYPE:
```

Comment (FlorianSW) [reply bundles/org.openhab.binding.groheondus/src/main/java/org/openhab/binding/groheondus/internal/discovery/GroheOndusDiscoveryService.java]: Agreed! :)

File: bundles/org.openhab.binding.groheondus/src/main/java/org/openhab/binding/groheondus/internal/handler/GroheOndusAccountHandler.java [hunk 2]
```diff
@@ -142,7 +142,9 @@ public void initialize() {
                 ondusService = OndusService.login(storage.get(STORAGE_KEY_REFRESH_TOKEN));
                 scheduleTokenRefresh();
             } else {
-                ondusService = OndusService.login(config.username, config.password);
+                // TODO: That's probably really inefficient, internally the loginWebform method acquires a refresh
+                // token, maybe there should be a way to obtain this token here, somehow.
+                ondusService = OndusService.loginWebform(config.username, config.password);
```
Comment (cpmeister) [inline bundles/org.openhab.binding.groheondus/src/main/java/org/openhab/binding/groheondus/internal/handler/GroheOndusAccountHandler.java]: This change is functionally equivalent though right?

Comment (FlorianSW) [reply bundles/org.openhab.binding.groheondus/src/main/java/org/openhab/binding/groheondus/internal/handler/GroheOndusAccountHandler.java]: Yes, it just uses another method to login :) From a user point of view, nothing really changes with that.

**Pull request merged**